tests/federation_room_event_auth_test.go

```diff
@@ -0,0 +1,206 @@
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/matrix-org/gomatrixserverlib"
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/federation"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestInboundFederationRejectsEventsWithRejectedAuthEvents(t *testing.T) {
+	/* These tests check that events which refer to rejected events in auth_events
+	 * are themselves rejected.
+	 *
+	 * In order to inject an outlier, we include it as an extra auth_event in a
```

Comment:
I've just tried to remind myself what an outlier is by looking in the spec. No matches that I could see. I'm guessing this isn't Synapse-specific but just something that arises when synchronising DAGs. Is it worth sticking a definition in the spec somewhere? A glossary even?

Reply:
There is https://matrix-org.github.io/synapse/latest/development/room-dag-concepts.html. Maybe these are synapse-specific terms/implementation choices after all?

Reply:
hrm. It's a grey area. It's not a concept that the spec ever refers to explicitly, since it tries not to define the exact algorithms that server implementations should use for these endpoints.

That said, I know that Dendrite uses a similar concept, and, though I've not looked at Conduit's implementation, a search for `outlier` brings up some likely-looking hits, so it probably does too.

So... it's not a thing that the spec defines, but it is a common concept to Matrix server implementations.

Every so often the idea of a "server implementation guide" gets raised (see https://github.com/matrix-org/matrix.org/tree/master/implementation-guides), and then nobody has time to work on them. That might be a natural home for defining some of these concepts.